src/axom/spin/internal/linear_bvh/build_radix_tree.hpp

```diff
@@ -464,13 +448,22 @@ void propagate_aabbs(RadixTree<FloatType, NDIMS>& data, int allocatorID)
 
   int32* counters_ptr = axom::allocate<int32>(inner_size, allocatorID);
 
+  using PointType = primal::Point<FloatType, NDIMS>;
+
+  PointType* min_range = axom::allocate<PointType>(inner_size, allocatorID);
+  PointType* max_range = axom::allocate<PointType>(inner_size, allocatorID);
+
   array_memset<ExecSpace>(counters_ptr, inner_size, 0);
+  array_memset<ExecSpace>(min_range, inner_size, PointType {FloatType {0}});
+  array_memset<ExecSpace>(max_range, inner_size, PointType {FloatType {0}});
```

Comment:
With the `atomicAdd`s for loads and stores, this is intentional. Since only one thread will handle each inner node `i`, there's only one `atomicAdd` issued for each point in `min_range` and`max_range`. So after an `atomicAdd(min_range[i][idim], value)` call, `min_range[i][idim] == value`.